src/pykeen/training/training_loop.py

```diff
@@ -389,10 +389,16 @@ def _train(  # noqa: C901
         # Take the biggest possible training batch_size, if batch_size not set
         batch_size_sufficient = False
         if batch_size is None:
-            if self.automatic_memory_optimization:
+            # Using automatic memory optimization on CPU may result in undocumented crashes due to OS' OOM killer.
+            if self.model.device.type == 'cpu':
+                batch_size = 256
+                batch_size_sufficient = True
+                logger.info(f"No batch_size provided. Setting batch_size to '{batch_size}'.")
```

Comment:
maybe better to inform the user that since it's on CPU, AMO won't get run, even if they set it to true

Reply:
I added some info statements.